corehq/util/models.py

```diff
@@ -241,3 +240,65 @@ def to_python(self, value):
     def pre_init(self, value, obj):
         value = super(NullJsonField, self).pre_init(value, obj)
         return self.get_default() if value is None else value
+
+
+class ForeignValue:
+    """Property descriptor for Django foreign key refs with a primitive value
+
+    This is useful for cases where the object referenced by the foreign
+    key holds a single primitive value named `value`. It eliminates
+    boilerplate indirection imposed by the foreign key, allowing the
+    value to be referenced as a simple attribute.
```

Comment:
Maybe this is such a common pattern that it would be obvious to many people, but I think if I hadn't already had the conversation with you about it, I wouldn't know that a primary use case of this would be to save space when there's a fairly small set of values but each value is somewhat long. Might be worth just spelling that out in this docstring for anyone who might not already be familiar with that pattern

Reply:
05f16bf6c3b52a70c14b9800539aa0af83a0e5fd